coherent_rates/system.py

```diff
@@ -72,128 +70,276 @@
-        EvenlySpacedTransformedPositionBasis[int, int, Literal[0], Literal[1]]
+        *tuple[EvenlySpacedTransformedPositionBasis[Any, Any, Any, Any], ...]
     ]
 ]:
-    interpolated = _get_interpolated_potential(system, resolution)
-    old = interpolated["basis"][0]
-    basis = TupleBasis(
-        EvenlySpacedTransformedPositionBasis[int, int, Literal[0], Literal[1]](
-            old.delta_x * shape[0],
-            n=old.n,
-            step=shape[0],
-            offset=0,
+    extrapolated_basis = TupleBasis(
+        *tuple(
+            EvenlySpacedTransformedPositionBasis[Any, Any, Any, Any](
+                old.delta_x * s,
+                n=old.n,
+                step=s,
+                offset=0,
+            )
+            for (old, s) in zip(
+                cast(Iterator[BasisWithLengthLike[Any, Any, Any]], potential["basis"]),
+                shape,
+            )
         ),
     )
-    scaled_potential = interpolated["data"] * np.sqrt(basis.fundamental_n / old.n)
 
-    return {"basis": basis, "data": scaled_potential}
+    scaled_potential = potential["data"] * np.sqrt(
+        extrapolated_basis.fundamental_n / potential["basis"].n,
+    )
 
+    return {"basis": extrapolated_basis, "data": scaled_potential}
 
-def get_potential(
-    system: PeriodicSystem,
-    config: PeriodicSystemConfig,
+
+def get_potential_1d(
+    system: PeriodicSystem1D,
+    shape: tuple[int, ...],
+    resolution: tuple[int, ...],
 ) -> Potential[
     TupleBasisWithLengthLike[
-        EvenlySpacedTransformedPositionBasis[int, int, Literal[0], Literal[1]]
+        *tuple[EvenlySpacedTransformedPositionBasis[Any, Any, Any, Any], ...]
     ]
 ]:
-    return _get_extended_potential(system, config.shape, config.resolution)
+    potential = _get_fundamental_potential_1d(system)
+    interpolated = _get_interpolated_potential(potential, resolution)
+
+    return _get_extrapolated_potential(interpolated, shape)
+
+
+def get_potential_2d(
+    system: PeriodicSystem2D,
+    shape: tuple[_L0Inv, ...],
+    resolution: tuple[int, ...],
+) -> Potential[
+    TupleBasisWithLengthLike[
+        *tuple[EvenlySpacedTransformedPositionBasis[Any, Any, Any, Any], ...]
+    ]
+]:
+    """Generate potential for 2D periodic system, for 111 plane of FCC lattice.
+
+    Expression for potential from:
+    [1] D. J. Ward
+        A study of spin-echo lineshapes in helium atom scattering from adsorbates.
+    [2]S. P. Rittmeyer et al
+        Energy Dissipation during Diffusion at Metal Surfaces:
+        Disentangling the Role of Phonons vs Electron-Hole Pairs.
+    """
+    potential = _get_fundamental_potential_2d(system)
+    interpolated = _get_interpolated_potential(potential, resolution)
+    return _get_extrapolated_potential(interpolated, shape)
 
 
 def _get_full_hamiltonian(
     system: PeriodicSystem,
-    shape: tuple[_L0Inv],
-    resolution: tuple[int],
+    shape: tuple[_L0Inv, ...],
+    resolution: tuple[_L0Inv, ...],
```

Comment:
_L1Inv, or just tuple[int, ...]. They don't need to be the same integer as shape (which this annotation suggests)